modules/utxo-lib/src/bitgo/UtxoPsbt.ts

```diff
@@ -317,3 +376,145 @@ export class UtxoPsbt<Tx extends UtxoTransaction<bigint>> extends Psbt {
     return { hash, sighashType };
   }
 }
+
+/**
+ * Takes a partially signed transaction and removes the scripts and signatures.
+ *
+ * Inputs must be one of:
+ *  * P2PKH
+ *  * P2SH 2-of-3
+ *  * P2WSH 2-of-3
+ *  * P2WPKH
+ *  * P2SH -> P2WSH 2-of-3
+ *  * P2SH -> P2WPKH
+ *  * P2TR script path 2-of-2
+ *
+ * @param tx the partially signed transaction
+ *
+ * @return the removed scripts and signatures, ready to be added to a PSBT
+ */
+export function unsign(tx: UtxoTransaction<bigint>, prevOuts: TxOutput<bigint>[]): PsbtInputUpdate[] {
+  const ret = tx.ins.map((input, vin) => {
+    const prevOut = prevOuts[vin];
+    const update: PsbtInputUpdate = {};
+    let redeemScript;
+    if (input.script && input.script.length) {
+      const decompiledScriptSig = bscript.decompile(input.script);
+      if (!decompiledScriptSig) {
+        throw new Error('Invalid scriptSig, failed to decompile');
+      }
+      if ((!input.witness || !input.witness.length) && decompiledScriptSig.length === 2) {
+        // P2PKH
```

Comment:
Should this be dropped since we don't support P2PKH on bitgo wallets (ditto for P2WPKH)? Although it probably doesn't hurt to have just for completeness sake since it's already implemented.

As an aside, do we need to cover P2SH-P2PK here since it's what we use for replay protection inputs?

Reply:
Yeah, these are useful in testing, which is why they're here.

We have a ticket open to support P2SH->P2PK. Thanks!